### PR TITLE
3 create the preprocessing functionality

### LINKED ISSUE #3: Create the preprocessing functionality

### PR DESCRIPTION
**Pull request by jorndijk**
- Added the required functions for preprocessing

### CODE REVIEW

File: src/lib_ml/preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        corpus.append(review)
+    return corpus
+
+def transform_data(dataset: pd.DataFrame, corpus: list) -> pd.DataFrame:
+    """
+    Transform the data into a format suitable for machine learning.
+    """
+    cv = CountVectorizer(max_features = 1420)
+    X = cv.fit_transform(corpus).toarray()
+    y = dataset.iloc[:, -1].values
```
Comment (kevinhoxhaa) [inline src/lib_ml/preprocessing.py]: During inference we do not have a label column, so I think `transform_data` should instead be in `model_service` and `model_training`, because the function will be different in each (training vs inference).

**Review by kevinhoxhaa — CHANGES_REQUESTED** [review]

File: src/lib_ml/processor.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.feature_extraction.text import CountVectorizer
+from lib_ml.preprocessing import create_corpus
+import numpy as np
+
+
+class RestaurantSentimentPreprocessor(BaseEstimator, TransformerMixin):
```
Comment (kevinhoxhaa) [inline src/lib_ml/processor.py]: Wouldn't it better to have this class in `model_training`? In the assignment they say lib_ml should be used just to preprocess text.

**Review by kevinhoxhaa — CHANGES_REQUESTED** [review]

File: src/lib_ml/preprocessing.py (hunk 1 above)
Comment (jorndijk) [reply src/lib_ml/preprocessing.py]: Yes I was wondering about that. A problem that arises with doing it separately is the making of the CountVectorizer model that vectorizes the input texts. This model should be used the same for the training and the prediction, so my initial thought is now to put it in the lib_ml package and just use it there. The other option would be to generate it in the model_training and ship it as a separate file to somewhere from there. What do you think?

File: src/lib_ml/processor.py (hunk 2 above)
Comment (jorndijk) [reply src/lib_ml/processor.py]: Yes, if we decide that the transform functionality should be split, then yes. But for now I thought if all transformation and cleaning is in this package, then it would be nice to add a class that can be used in a ML pipeline.

File: src/lib_ml/preprocessing.py (hunk 1 above)
Comment (kevinhoxhaa) [reply src/lib_ml/preprocessing.py]: I think the best way to do it is to generate it in the `model_training `and ship it as a separate file to somewhere from there. So basically I should be able to query a URL that you provide that will let `model-service` download both the model itself and the CountVectorizer as pkl.

File: src/lib_ml/processor.py (hunk 2 above)
Comment (kevinhoxhaa) [reply src/lib_ml/processor.py]: Yeah, so I think we should go with the solution I proposed above so we split the transform functionality, and thus move this to `model_training`.

File: src/lib_ml/preprocessing.py (hunk 1 above)
Comment (jorndijk) [reply src/lib_ml/preprocessing.py]: Lib_ml should be good now

**Review by kevinhoxhaa — APPROVED** [review]
Nice changes!

**Review by flazedd — APPROVED** [review]
Clean code, well structured!
